server/plugin.go

```diff
@@ -553,6 +554,56 @@ func (p *Plugin) sendRefreshEvent(userID string) {
 	)
 }
 
+func (p *Plugin) sendChannelSubscriptionsUpdated(channelID string) {
+	config := p.getConfiguration()
+	gitlabURL, err := url.Parse(config.GitlabURL)
+	if err != nil {
+		p.API.LogWarn(
+			"unable to parse GitlabURL",
+			"err", err.Error(),
+		)
+		return
+	}
+
+	subscriptions, err := p.GetSubscriptionsByChannel(channelID)
+	if err != nil {
+		p.API.LogWarn(
+			"unable to fetch subscriptions by channel",
+			"err", err.Error(),
+		)
+		return
+	}
+
+	var payload struct {
+		ChannelID     string                 `json:"channel_id"`
+		Subscriptions []SubscriptionResponse `json:"subscriptions"`
+	}
+	payload.ChannelID = channelID
+	payload.Subscriptions = make([]SubscriptionResponse, 0, len(subscriptions))
+
+	for _, subscription := range subscriptions {
+		payload.Subscriptions = append(payload.Subscriptions, SubscriptionResponse{
+			RepositoryName: subscription.Repository,
+			RepositoryURL:  gitlabURL.JoinPath(subscription.Repository).String(),
+		})
+	}
+
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		p.API.LogWarn(
+			"unable to marshal payload for updated channel subscriptions",
+			"err", err.Error(),
+		)
+		return
+	}
+
+	p.API.PublishWebSocketEvent(
+		WsChannelSubscriptionsUpdated,
+		map[string]interface{}{"payload": string(payloadJSON)},
+		&model.WebsocketBroadcast{ChannelId: channelID},
```

Comment:
This constrains the websocket event to members of the channel, /but/ I see there is logic in the plugin around some users being denied access to some subset of projects. Worse, it looks like these permission checks actually involve a round trip per user -- is there some background information I can learn about?

Reply:
IIUC, but isn't it implied that access to the channel gives access to the subscription information? 
So shouldn't this be left to channel admins (by controlling channel memberships)?

Reply:
@lieut-data It looks like the `/gitlab subscriptions list` command does not filter on any sort of permission checks, which is sort of the equivalent of this websocket message right? I'm thinking we may not need any further access control other than scoping it to the specific channel, as you've done here.

Reply:
Awesome, thank you for this feedback -- will leave this as-is.